commit 25c70097a65a22a5799442c8935fa927d91bddaf
Author: Kent Overstreet <kent.overstreet@linux.dev>
Date:   Sun Jun 11 18:24:04 2023 -0400

    bcachefs: Delete weird hacky transaction restart injection
    
    since we currently don't have a good fault injection library,
    bch2_btree_insert_node() was randomly injecting faults based on
    local_clock().
    
    At the very least this should have been a debug mode only thing, but
    this is a brittle method so let's just delete it.
    
    Signed-off-by: Kent Overstreet <kent.overstreet@linux.dev>

diff --git a/fs/bcachefs/btree_update_interior.c b/fs/bcachefs/btree_update_interior.c
index 66da1da2f075..d2811c4756b7 100644
--- a/fs/bcachefs/btree_update_interior.c
+++ b/fs/bcachefs/btree_update_interior.c
@@ -1674,9 +1674,6 @@ static int bch2_btree_insert_node(struct btree_update *as, struct btree_trans *t
 	BUG_ON(!as || as->b);
 	bch2_verify_keylist_sorted(keys);
 
-	if ((local_clock() & 63) == 63)
-		return btree_trans_restart(trans, BCH_ERR_transaction_restart_split_race);
-
 	ret = bch2_btree_node_lock_write(trans, path, &b->c);
 	if (ret)
 		return ret;